### ISSUE TITLE
Wrong timezone / timezone can not be set

### ISSUE THREAD

**Issue by TheMagican** [ehough/docker-kodi#5]
# Bug description
When executing erichough/kodi:latest, the time displayed in the upper right corner of Kodi is always UTC. In the Kodi settings, it is not possible to change the timezone to the correct values, as the options for this are disabled and can not be changed. The time is in UTC despite the fact, that the correct timezone is set in the container.

# Steps to reproduce
- Set the timezone of your computer to something else than UTC. In my case `Europe/Berlin`
- Start kodi: `x11docker --hostdisplay --pulseaudio --wm none --gpu --homedir /some/dir/kodi-home -- erichough/kodi:latest`
- In Kodi, the time displayed in the upper right corner is wrong (in my case, it is minus two hours)
- Open a new terminal while Kodi is running and execute `docker exec -ti $(docker ps | grep erichough/kodi | awk '{ print $1 }') date` to get the time in the container. The time displayed there is the same than the host machine's time

# What I did to isolate the issue
## Compare to local Kodi installation
I installed Kodi on a fresh Ubuntu bionic installation (outside of a docker container): The time displayed there is correct. Additionally, the timezone settings are **not** disabled and may be changed.

See attached screenshots of the dockerized Kodi and a native bionic installation:
- Displayed time differs by 2h
- Timezone settings are editable in native installation and disabled in container

![kodi_docker](https://user-images.githubusercontent.com/4193540/41004241-842727e8-6919-11e8-9869-db92908048a0.jpg)
![kodi_virtualbox](https://user-images.githubusercontent.com/4193540/41004242-8448f5b2-6919-11e8-8deb-b73f3782e887.jpg)

## Explicitely set the timezone in container
I started the docker container with `"-v /etc/timezone:/etc/timezone:ro" "-e TZ=$(cat /etc/timezone)"` to both mount `/etc/timezone` of the host computer and set the `TZ` environment variable - which should both set the correct timezone. This didn't change anything.

## Compare to working container
I cloned the repo and reverted back to commit 367f17417a4e8a9e3169993cf78b871c83e1973a. I then built the container from this commit and found, that Kodi displayed the correct time. Simply changing `FROM ubuntu:xenial` to `FROM ubuntu:bionic` caused the error to appear causing the time to be wrong.

**Comment by ehough** [comment]
Thanks for this report!

Funny thing - last week I noticed the same thing on my Kodi installation and forgot all about it until now. So I can confirm that it's not just you that's experiencing this bug.

> Simply changing FROM ubuntu:xenial to FROM ubuntu:bionic caused the error to appear causing the time to be wrong.

That's a huge clue, thanks. I'll investigate and report back here once I have a solution. Stand by..

**Comment by ehough** [comment]
tl;dr add `-v /usr/share/zoneinfo:/usr/share/zoneinfo:ro` to your command.

I looked through the Kodi source code and [found](https://github.com/xbmc/xbmc/blob/07b5e98a1fbea5057e4021154b52497c62099d34/xbmc/platform/linux/LinuxTimezone.cpp) that it relies on the `/usr/share/zoneinfo` directory, which is not present in the container. Bind-mounting it into the Kodi container enables the Timezone country (Zeitzonen-Region for the Deutchlanders :wink:). Once a country is selected, the Timezone (Zeitzonen) dropdown is enabled.

I'm going to open an issue with `x11docker` to get his thoughts on this.

**Comment by TheMagican** [comment]
@ehough : Thanks for this fast solution!

I can confirm, that adding `-v /usr/share/zoneinfo:/usr/share/zoneinfo:ro` to the x11docker commandline both makes kodi display the correct timezone and also makes the timezone settings in kodi editable.
